src/lib/utils.ts

```diff
@@ -0,0 +1,124 @@
+import type { JSONSchema, SchemaObject } from 'oas/dist/rmoas.types';
+
+import { get as lodashGet } from 'lodash'; // eslint-disable-line no-restricted-imports
+
+/**
+ * Determine if a schema `type` is, or contains, a specific discriminator.
+ *
+ */
+export function hasSchemaType(
+  schema: SchemaObject,
+  discriminator: 'array' | 'object' | 'string' | 'number' | 'boolean' | 'integer' | 'null'
+) {
+  if (Array.isArray(schema.type)) {
+    return schema.type.includes(discriminator);
+  }
+
+  return schema.type === discriminator;
+}
+
+/**
+ * With a supplied JSON Schema object, spider through it for any schemas that may contain specific
+ * kind of `format` that also happen to be within the current `requestBody` payload that we're
+ * creating a HAR representation for.
+ *
+ */
+export function getTypedFormatsInSchema(
```

Comment:
this block of code is hard to parse through. maybe breaking it up into functions will make it easier to read.

```suggestion

function handleParentDataArrays(parentData: any, opts: { parentIsArray?: boolean; parentKey?: string; payload: any }) {
  return Object.keys(parentData)
    .map(pdk => {
      const currentKey = [opts.parentKey, pdk /* , key */].join('.');
      if (lodashGet(opts.payload, currentKey) !== undefined) {
        return currentKey;
      }

      return false;
    })
    .filter(Boolean);
}

function getSubSchemas(subSchemaDataSize: number, schema: any, opts: {
  parentIsArray?: boolean;
  parentKey?: string;
  payload: any
}) {
  let subSchemas: any[] = [];
  if (subSchemaDataSize > 0) {
    for (let idx = 0; idx < subSchemaDataSize; idx += 1) {
      subSchemas = subSchemas.concat(
        Object.entries(schema).map(([key, subSchema]: [string, JSONSchema]) => ({
          key: opts.parentKey ? [opts.parentKey, idx, key].join('.') : key,
          schema: subSchema,
        }))
      );
    }
  } else {
    return Object.entries(schema).map(([key, subSchema]: [string, JSONSchema]) => ({
      key: opts.parentKey ? [opts.parentKey, key].join('.') : key,
      schema: subSchema,
    }));
  }
}

/**
 * With a supplied JSON Schema object, spider through it for any schemas that may contain specific
 * kind of `format` that also happen to be within the current `requestBody` payload that we're
 * creating a HAR representation for.
 *
 */
export function getTypedFormatsInSchema(
  format: 'json' | 'binary',
  schema: any,
  opts: {
    parentIsArray?: boolean;
    parentKey?: string;
    payload: any;
  }
): boolean | string | (string | boolean)[] {
  try {
    if (schema?.format === format) {
      const parentData = lodashGet(opts.payload, opts.parentKey);
      if (opts.parentIsArray && parentData !== undefined && Array.isArray(parentData)) {
        return handleParentDataArrays(parentData, opts);
      } else if (opts.parentKey && parentData !== undefined) {
        return opts.parentKey;
      } else if (opts.payload !== undefined) {
        // If this payload is present and we're looking for a specific format then we should assume
        // that the **root** schema of the request body is that format, and we aren't trafficking in
        // a nested object or array schema.
        return true;
      }

      return false;
    }

    let subSchemaDataSize = 0;
    if (opts.parentIsArray) {
      // If we don't have data for this parent schema in our body payload then we
      // shouldn't bother spidering further into the schema looking for more `format`s
      // for data that definitely doesn't exist.
      const parentData = lodashGet(opts.payload, opts.parentKey);
      if (parentData === undefined || !Array.isArray(parentData)) {
        return false;
      }

      subSchemaDataSize = parentData.length;
    }
    const subSchemas = getSubSchemas(subSchemaDataSize, schema, opts);

    return subSchemas
      .map(({key, schema: subSchema}) => {
        if ('properties' in subSchema) {
          return getTypedFormatsInSchema(format, subSchema.properties, {payload: opts.payload, parentKey: key});
        }

        if ('items' in subSchema) {
          const schema = (subSchema.items as JSONSchema)?.properties ?? subSchema.items;

          return getTypedFormatsInSchema(format, schema, {
            payload: opts.payload,
            parentKey: key,
            parentIsArray: true,
          });
        }

        // If this schema has neither `properties` or `items` then it's a regular schema we can re-run.
        return getTypedFormatsInSchema(format, subSchema, {payload: opts.payload, parentKey: key});
      })
      .flat()
      .filter(Boolean);
  } catch (err) {
    // If this fails for whatever reason then we should act as if we didn't find any `format`'d
    // schemas.
    return [];
  }
}
```

Reply:
Good idea